src/compiler/scala/tools/nsc/transform/OverridingPairs.scala

```diff
@@ -24,6 +25,11 @@ import scala.reflect.internal.SymbolPairs
 abstract class OverridingPairs extends SymbolPairs {
   import global._
 
+  @nowarn("""cat=deprecation&origin=scala\.tools\.nsc\.transform\.OverridingPairs\.Cursor""")
+  final type PairsCursor = Cursor
+
+  @nowarn("msg=shadowing a nested class of a parent is deprecated")
+  @deprecated("use PairsCursor instead", since = "2.13.4")
   class Cursor(base: Symbol) extends super.Cursor(base) {
```

Comment:
This one would also prevent the Scala.js compiler plugin to compile without warnings (it currently has -Xfatal-warnings, and I very much hope to keep it that way).